src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -229,7 +229,7 @@ private bool DetermineExtensionDataProperty(Dictionary<string, JsonPropertyInfo>
                     typeof(IDictionary<string, JsonElement>).IsAssignableFrom(declaredPropertyType))
                 {
                     JsonConverter? converter = Options.GetConverter(declaredPropertyType);
-                    if (converter == null)
+                    if (converter == null || jsonPropertyInfo.RuntimeClassInfo.CreateObject == null)
```

Comment:
https://github.com/dotnet/runtime/blob/4fd3c8cc6052cf11ccd1ecdc36331164bef41992/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs#L121 would be a better place for this check. We shouldn't throw unless there's extension data to store.

We'll need tests for both cases, showing that deserialization works fine when there's no extension data, and one showing the `NSE` when theres extra data in the payload.

Reply:
Got it -- that was my first instinct when I ran this through the debugger, but then I saw that other exceptions for this property's type were thrown elsewhere so I second-guessed myself.  Will make this change.

Reply:
Fixed in 886316c

Reply:
It might make sense move those checks to the point of (de)serialization if they can be done in way that doesn't hurt runtime performance. This can be considered as part of https://github.com/dotnet/runtime/issues/32903.